torch/distributed/algorithms/ddp_comm_hooks/ddp_zero_hook.py

```diff
@@ -385,11 +386,12 @@ def hook_with_zero_step_interleaved(
     ddp_ref = weakref.ref(ddp)
 
     # NOTE: Gloo may hang with this overlapping approach, so we require
-    # NCCL backend for now; see https://github.com/pytorch/pytorch/issues/62300
-    if dist.get_backend(ddp_ref().process_group) != dist.Backend.NCCL:  # type: ignore[union-attr]
+    # NCCL/HCCL backend for now; see https://github.com/pytorch/pytorch/issues/62300
+    if ((dist.get_backend(ddp_ref().process_group) != dist.Backend.NCCL) and
+        (dist.get_backend(ddp_ref().process_group) != 'hccl')):  # type: ignore[union-attr]
```

Comment:
not sure if we want to hardcode process group impls like this, but I guess it does follow existing convention.

We could for now have an allowlist such as:

```
BACKEND_ZERO_OVERLAP_SUPPORTED = [nccl]
if hccl_available():
    BACKEND_ZERO_OVERLAP_SUPPORTED.append(hccl)
```

cc @awgu

Reply:
@rohan-varma we were not able to add hpu backend previously. So am not sure if there is a way to check for the availability of hccl or hpu backend with the python interface in fork. Ref: https://github.com/pytorch/pytorch/pull/71397

Reply:
@mrshenli Based on the discussions above, is it okay to include this sort of backend hardcoding here for now?